src/main/java/org/jenkins/plugins/lockableresources/actions/LockableResourcesRootAction.java

```diff
@@ -126,6 +126,58 @@ public void doReserve(StaplerRequest req, StaplerResponse rsp)
 		rsp.forwardToPreviousPage(req);
 	}
 
+	public void doReassign(StaplerRequest req, StaplerResponse rsp)
+		throws IOException, ServletException {
+		Jenkins.getInstance().checkPermission(RESERVE);
+
+		String name = req.getParameter("resource");
+		LockableResource r = LockableResourcesManager.get().fromName(name);
+		if (r == null) {
+			rsp.sendError(404, "Resource not found " + name);
+			return;
+		}
+
+		String userName = getUserName();
+		if ( userName == null ||
+				!Jenkins.getInstance().hasPermission(Jenkins.ADMINISTER))
```

Comment:
This is another permission check. What do you want to achieve here? Either lump this under the "RESERVE" permission object (maybe we need to redefine it? Renaming it is probably out of the question, since it's already in use) or create another permission object to guard this new operations...

Reply:
Like commented elsewhere, this is a verbatim copy of logic from another button handler nearby, trusting the original authors knew their stuff :)

Reply:
I hope this was rectified in my later summer commits